File: library/src/main/java/com/licrafter/tagview/views/ITagView.java
package com.licrafter.tagview.views;

import com.licrafter.tagview.DIRECTION;

/**
 * author: shell
 * date 2016/12/22 下午1:04
 **/

public interface ITagView {

    // set tag direction
    void setDirection(DIRECTION direction);

    // get tag dirction
    DIRECTION getDirection();

    int getMeasuredWidth();

    int getMeasuredHeight();

    int getTop();

    int getLeft();

    int getRight();

    int getBottom();

    void layout(int left, int top, int right, int bottom);

}

File: library/src/main/kotlin/com/licrafter/tagview/views/ITagView.kt
package com.licrafter.tagview.views

import com.licrafter.tagview.DIRECTION

/**
 * Created by lijx on 2018/5/7.
 * Gmail: shellljx@gmail.com
 */
interface ITagView {

    // set tag direction
    fun setDirection(direction: DIRECTION)

    // get tag dirction
    fun getDirection(): DIRECTION

    fun getMeasuredWidth(): Int

    fun getMeasuredHeight(): Int

    fun getTop(): Int

    fun getLeft(): Int

    fun getRight(): Int

    fun getBottom(): Int

    fun layout(left: Int, top: Int, right: Int, bottom: Int)

}